lib/src/rules/avoid_types_as_parameter_names.dart

```diff
@@ -46,17 +47,28 @@ class _Visitor extends SimpleAstVisitor<void> {
 
   _Visitor(this.rule);
 
+  bool hasTypeName(String name) =>
```

Comment:
'hasTypeName' --> 'isTypeName'

This seems very prone to false positives. I'd like to see us add a method to `LinterContext` (or someplace better) that would allow us to perform a name lookup at a given point in the AST. This method should then be rewritten to perform that lookup (in the parent of the declaration site so we don't get back the existing declaration) and see whether the name is bound to a type.

Reply:
> 'hasTypeName' --> 'isTypeName'

👍 

As for false positives, totally agreed.  The current heuristic is a kludge.  Adding a comment for the `LinterContext` idea.

fyi @scheglov: who might have ideas/reservations.

Reply:
There is already `LinterContext.resolveNameInScope()`, which probably will help here.
I will need to try to switch this lint to it.

Reply:
Ah, cool!

Reply:
https://github.com/dart-lang/linter/pull/2062